src/psyclone/psyir/backend/c.py

```diff
@@ -56,10 +59,71 @@
                  ScalarType.Intrinsic.REAL: "double"}
 
 
-class CWriter(PSyIRVisitor):
+class CWriter(LanguageWriter):
     '''Implements a PSyIR-to-C back-end for the PSyIR AST.
 
+    :param bool skip_nodes: If skip_nodes is False then an exception \
+        is raised if a visitor method for a PSyIR node has not been \
+        implemented, otherwise the visitor silently continues. This is an \
+        optional argument which defaults to False.
+    :param indent_string: Specifies what to use for indentation. This \
+        is an optional argument that defaults to two spaces.
+    :type indent_string: str or NoneType
+    :param int initial_indent_depth: Specifies how much indentation to \
+        start with. This is an optional argument that defaults to 0.
+    :param bool check_global_constraints: whether or not to validate all \
+        global constraints when walking the tree.
+
+    :raises TypeError: if any of the supplied parameters are of the wrong type.
+
     '''
+    def __init__(self, skip_nodes=False, indent_string="  ",
+                 initial_indent_depth=0, check_global_constraints=True):
+
+        super(CWriter, self).__init__(["[", "]"], ".", skip_nodes,
+                                      indent_string,
+                                      initial_indent_depth,
+                                      check_global_constraints)
+
+    def gen_dims(self, shape, var_name=None):
+        '''Given a list of PSyIR nodes representing the dimensions of an
+        array, return a list of strings representing those array dimensions.
```

Comment:
After writing the previous comment I realized that in Fortran they are called dimension and shape because in fact it is also used to generate the declaration expressions. But this seems more of a convenience of Fortran so I would still recommend the renaming and leave a comment in the Fortran docstring version of this method that it also works if PSyIR shapes instead of indices are provided.

Reply:
And I read the last comment only after I had done everything else :) But all done.